core/src/main/java/org/apache/brooklyn/core/catalog/internal/BasicBrooklynCatalog.java

```diff
@@ -855,24 +893,50 @@ private boolean attemptType(String key, CatalogItemType candidateCiType) {
             }
             // first look in collected items, if a key is given
             String type = (String) item.get("type");
-            String version = null;
-            if (CatalogUtils.looksLikeVersionedId(type)) {
-                version = CatalogUtils.getVersionFromVersionedId(type);
-                type = CatalogUtils.getSymbolicNameFromVersionedId(type);
-            }
+            
             if (type!=null && key!=null) {
                 for (CatalogItemDtoAbstract<?,?> candidate: itemsDefinedSoFar) {
                     if (candidateCiType == candidate.getCatalogItemType() &&
                             (type.equals(candidate.getSymbolicName()) || type.equals(candidate.getId()))) {
-                        if (version==null || version.equals(candidate.getVersion())) {
-                            // matched - exit
-                            catalogItemType = candidateCiType;
-                            planYaml = candidateYaml;
-                            resolved = true;
-                            return true;
+                        // matched - exit
+                        catalogItemType = candidateCiType;
+                        planYaml = candidateYaml;
+                        resolved = true;
+                        return true;
+                    }
+                }
+            }
+            {
+                // legacy routine; should be the same as above code added in 0.12 because:
+                // if type is symbolic_name, the type will match below, and version will be null so any version allowed to match 
```

Comment:
grand, but how about the question above where type is of form symbolic_name:version? When you say "the id will match", where does that happen?